File: 1458-sort-integers-by-the-number-of-1-bits/sort-integers-by-the-number-of-1-bits.js
/**
 * @param {number[]} arr
 * @return {number[]}
 */
var sortByBits = function(arr) {
    const obj = {}
    for(let i of arr){
        let b = i.toString(2)
        let c = b.split("").filter((n)=>n==1).length
        if(!(c in obj)){
            obj[c] = [i]  
        }else{
            obj[c].push(i)
        }
    }
    let sortedKeys = Object.keys(obj).sort((a, b) => Number(a) - Number(b));
    let ans = []
    for(let i of sortedKeys){
        ans = [...ans, ...obj[i].sort((a,b)=>a-b)]
    }
    return ans
};

File: 1458-sort-integers-by-the-number-of-1-bits/sort-integers-by-the-number-of-1-bits.py
class Solution:
    def sortByBits(self, arr: List[int]) -> List[int]:
        d = {}
        for i in arr:
            c = bin(i).count("1")
            if c not in d:
                d[c] = [i]
            else:
                d[c].append(i)
        if len(d) == 1:
            for i in d:
                return sorted(d[i])
        ans = []
        for i in sorted(d.keys()):
            ans+=sorted(d[i])
        return ans